java/ql/src/semmle/code/java/security/Encryption.qll

```diff
@@ -222,9 +222,14 @@ class JavaSecurityMessageDigest extends JavaSecurityAlgoSpec {
     exists(Constructor c | c.getAReference() = this |
       c.getDeclaringType().getQualifiedName() = "java.security.MessageDigest"
     )
+    or
+    exists(Method m | m.getAReference() = this |
+      m.getDeclaringType().getQualifiedName() = "java.security.MessageDigest" and
```

Comment:
Actually, there's one minor thing to note here.  We generally prefer `.hasQualifiedName(.., ..)` over `.getQualifiedName()`, since the former doesn't rely on string construction in QL

Reply:
This means that `hasQualifiedName` can start the join chain from a single tuple and go directly to the right type, whereas `getQualifiedName` has to scan all types doing a string concatenation for each of them before selecting the right one.

Reply:
Thanks, I also updated the line above.